### PR TITLE
Improve use_tag_stack-hack: use context managers, short 'tags'

### PR DESCRIPTION
**Pull request by blueyed**
Use a NamedTemporaryFile context manager, which removes the temporary
file again, and use it as the only entry for the 'tags' setting to
improve lookup performance.

/cc @wilywampa

### CODE REVIEW

**Comment by wilywampa** [comment]
I actually used `NamedTemporaryFile` in my initial implementation: https://github.com/wilywampa/jedi-vim/commit/5041224c13f3f799db55c6d72137dbfde3a761f9

I changed to `tempfile()` because the Travis CI test didn't like `NamedTemporaryFile` but I didn't try it with a `prefix` argument, but that works so it's better.

If we're setting tags to just one file we can skip the while loop and addition of random characters to the tag name because there won't be any collisions by definition.

File: jedi_vim.py [hunk 1]
```diff
@@ -262,10 +263,13 @@ def goto(mode="goto", no_output=False):
                                  for _ in range(4)])
                         f.write('{0}\t{1}\t{2}'.format(tagname, d.module_path,
                             'call cursor({0}, {1})'.format(d.line, d.column + 1)))
-                        f.seek(0)
```
Comment (davidhalter) [inline jedi_vim.py]: @wilywampa @blueyed 
Do you guys know why this was there in the beginning? Why would you seek there?

Comment (blueyed) [inline jedi_vim.py]: Maybe to trigger a flush?

**Comment by blueyed** [comment]
> skip the while loop

Will add that.

File: jedi_vim.py (hunk 1 above)
Comment (wilywampa) [inline jedi_vim.py]: For this: http://stackoverflow.com/questions/1202848/python-tempfile-temporaryfile-cannot-be-read-why

Does `flush` do the same thing?

Comment (blueyed) [inline jedi_vim.py]: `flush` is necessary for the file to be readable/filled by Vim.
The above SO answer might be only relevant for the same Python process maybe (a comment states that `flush` does not work in this case)?!

**Comment by blueyed** [comment]
> If we're setting tags to just one file we can skip the while loop and addition of random characters to the tag name because there won't be any collisions by definition.

Done.

**Comment by davidhalter** [comment]
Is this one ready? (There are merge conflicts btw.)

**Comment by blueyed** [comment]
Rebased and amended it.
Should be good from my side.

@wilywampa ?

**Comment by blueyed** [comment]
I'm going ahead and merge it, hopefully fixing #482.

**Pull request merged**